libs/azure-storage/tests/unit_tests/test_document_loaders.py

```diff
@@ -0,0 +1,160 @@
+import azure.identity
+import pytest
+from azure.storage.blob import ContainerClient
+from pytest_mock import MockerFixture
+
+from langchain_azure_storage.document_loaders import AzureBlobStorageLoader
+
+
+@pytest.fixture
+def blob_names() -> list[str]:
+    return ["text_file.txt", "json_file.json", "csv_file.csv"]
+
+
+@pytest.fixture
+def blob_contents() -> list[str]:
+    return [
+        "test content",
+        "{'test': 'test content'}",
+        "col1,col2\nval1,val2",
+    ]
+
+
+def _blob_storage_loader_args() -> dict:
+    return {
+        "account_url": "https://testaccount.blob.core.windows.net",
+        "container_name": "test-container",
+    }
+
+
+@pytest.fixture
+def mock_lazy_load_documents_from_blob(
+    mocker: MockerFixture, blob_names: list[str], blob_contents: list[str]
+) -> None:
+    from langchain_core.documents.base import Document
+
+    docs = []
+    for name, content in zip(blob_names, blob_contents):
+        docs.append(
+            Document(
+                page_content=content,
+                metadata={
+                    "source": f"https://testaccount.blob.core.windows.net/test-container/{name}"
+                },
+            )
+        )
+    mocker.patch(
+        "langchain_azure_storage.document_loaders.AzureBlobStorageLoader._lazy_load_documents_from_blob",
+        return_value=iter(docs),
+    )
```

Comment:
Instead of patching the loader, we should instead be updating our mock container and blob clients to list and return the contents in the expected order. The main reason that I'm suggesting that is in bypassing the logic around downloading the blob contents and turning them into a document, we will have a gap in our tests as we technically would not be exercising that method any more since its been patched out.